RELEASE/scripts/autoscend/iotms/mr2021.ash

```diff
@@ -112,3 +112,238 @@ int auto_backupUsesLeft()
 {
 	return 11 + (my_path() == "You, Robot" ? 5 : 0) - get_property("_backUpUses").to_int();
 }
+
+boolean auto_havePowerPlant()
+{
+	return item_amount($item[potted power plant]) > 0 && auto_is_valid($item[potted power plant]);
+}
+
+boolean auto_harvestBatteries()
+{
+	if(!auto_havePowerPlant() || get_property("_pottedPowerPlant") == "0,0,0,0,0,0,0")
+	{
+  		return false;
+	}
+
+	// Stolen straight from mafia's breakfast handling.
+	cli_execute("inv_use.php?pwd&whichitem=" + $item[potted power plant].to_int());
+	
+	string [int] status = split_string(get_property("_pottedPowerPlant"), ",");
+
+	for ( int pp = 0; pp < status.count(); pp++ )
+	{
+		if ( status[pp] > 0)
+		{
+			cli_execute("choice.php?pwd&whichchoice=1448&option=1&pp=" + ( pp + 1 ));
+		}
+	}
+	return true;
+}
+
+// These points the value of a battery represented in AAAs.
+int batteryPoints(item battery)
+{
+	static int[item] points = {
+		$item[battery (AAA)]: 1,
+		$item[battery (AA)]: 2,
+		$item[battery (D)]: 3,
+		$item[battery (9-Volt)]: 4,
+		$item[battery (lantern)]: 5,
+		$item[battery (car)]: 6
+	};
+	return points[battery];
+}
+
+// These points represent a quantity of AAAs if all batteries were untinkered.
+int totalBatteryPoints()
+{
+	int totalPoints = 0;
+
+	foreach it in $items[battery (AAA), battery (AA), battery (D), battery (9-Volt), battery (lantern), battery (car)]
+	{
+		totalPoints += available_amount(it) * batteryPoints(it);
+	}
+
+	return totalPoints;
+}
+
+// Mafia's handling of the create() function is currently not flexible enough for crafting batteries.
+// This is very dense, apologies.
+boolean batteryCombine(item battery)
+{
+	if (!($items[battery (AAA), battery (AA), battery (D), battery (9-Volt), battery (lantern), battery (car)] contains battery))

```

Comment:
```suggestion
	if(batteryPoints(battery) == 0)	//0 means it is not a battery
```
@Malibu-Stacey what do you think of the above? since you suggested the contains method in the previous PR.

Reply:
Looks good assuming `batteryPoints` returns 0 for non-batteries.